pkg/datapath/linux/ipsec/ipsec_linux_test.go

```diff
@@ -82,6 +84,22 @@ func TestLoadKeys(t *testing.T) {
 	}
 }
 
+func TestLoadKeysIdentical(t *testing.T) {
+	log := setupIPSecSuitePrivileged(t)
+
+	keys := bytes.NewReader(keysDuplicateDat)
+	_, _, err := LoadIPSecKeys(log, keys)
+	require.NoError(t, err)
+}
+
+func TestLoadKeysSameSPI(t *testing.T) {
+	log := setupIPSecSuitePrivileged(t)
+
+	keys := bytes.NewReader(keysSameSpiDat)
+	_, _, err := LoadIPSecKeys(log, keys)
+	require.Error(t, err)
+}
```

Comment:
Isn't there an implicit dependency between these two test cases? If someone removes the first, or loads a new key in between, it will break the second even though the second won't have been modified. Or did I miss something?

Reply:
Right now I opted for introducing and leaving only the `TestLoadKeysSameSPI` test, which tests the actual use case when a new valid key has been provided without increasing the SPI. I think this test should be enough to logically test what is being introduced and addressed with this PR.